### ISSUE TITLE
Re-enable `reachability-slice` if CBMC fixes soundness issues

### ISSUE THREAD

**Issue by avanhatt** [model-checking/kani#683]
In https://github.com/diffblue/cbmc/issues/6394, we found that CBMC could unsoundly change results with certain slicing flags. CBMC now has in progress PRs to address some of the root issues, when those land, we should consider re-adding slicing flags for performance. For one proof they had a change on the order of ~10 minutes faster.

**Comment by zhassan-aws** [comment]
Issue https://github.com/diffblue/cbmc/issues/6394 is fixed, but this is now blocked by a new issue: https://github.com/diffblue/cbmc/issues/6730.

**Comment by tautschnig** [comment]
I've tried to provide guidance on https://github.com/diffblue/cbmc/issues/6730 - I think changes in kani's sequence of calls are required to address this.

**Comment by zhassan-aws** [comment]
We found that `--slice-formula` performs better, and it's now enabled by default. Closing this one.
